### ISSUE TITLE
Log_probabilities returned by tf.nn.ctc_beam_search_decoder

### ISSUE THREAD

**Issue by karanMal** [tensorflow/tensorflow#21120]
**System Info**

- Have I written custom code=Yes
- OS Platform and Distribution :: Linux ( CentOS version 6.6 )
- TensorFlow installed from: binary (pip install)
- TensorFlow version : 1.3.0
- Bazel version: N/A
- CUDA/cuDNN version : v8.0.44
- GPU model and memory: NVIDIA Quadro P5000 (16272MiB)
- Exact command to reproduce :N/A 
- Mobile device: N/A

**Problem**
I am training a LSTM-CTC speech recognition system with using beam search decoding in the following configuration:

`decoded, log_prob =
tf.nn.ctc_beam_search_decoder(
    inputs,
    sequence_length,
    beam_width=100,
    top_paths=3,
    merge_repeated=True
)`

The output of log_probabilities for a batch by the above decoder are like-
 
`     

       [ [ 20.45407486,  20.44991684,  20.41798401],

        [ 14.9961853 ,  14.925807  ,  14.88066769],
        ..., 
        [ 18.89863396,  18.85992241,  18.85712433],

        [  3.93567419,   3.92791557,   3.89198923],

        [ 14.56258488,  14.55923843,  14.51092243]],`

So how these scores represent log probabilities and if I want to compare confidence for top paths among examples then what will be the normalisation factor???

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by drpngx** [comment]
Unfortunately, you don't have a probability, because it would mean summing over all possible sequences of all admissible lengths. You can pretend that the top hypotheses that you get out of the decoder comprise all of the probability mass. In that case, then the scores are in log domain. You can compute

```
Z = sum_k(exp(score_k))
```
Then each sequence has a probability of `exp(score_k) / Z`.
